packages/descendants/src/useDescendant.tsx

```diff
@@ -0,0 +1,80 @@
+import {
+  ComponentProps,
+  ForwardedRef,
+  RefObject,
+  useContext,
+  useMemo,
+  useRef,
+} from 'react';
+
+import {
+  useForwardedRef,
+  useIsomorphicLayoutEffect,
+} from '@leafygreen-ui/hooks';
+
+import { DescendantContextType } from './DescendantsContext';
+import { findDescendantIndexWithId } from './utils';
+
+/** @deprecated */
+const genId = () => `_${Math.random().toString(36).substr(2, 9)}`;
```

Comment:
why deprecated? should we use our idAllocator here?

Reply:
Ah, I wanted to add a bit of a more robust id generator, but this _should_ be fine.
We may want to add some collision-prevention logic